### PR TITLE
Fix packaging bash scripts

### PR DESCRIPTION
**Pull request by TaoChenOSU**
### Motivation and Context

Running the packaging bash scripts in the repo will result in the following error:
image
This is caused by un-escaped characters in the build options.

### Description
Fix the issue by escaping the characters in the build options.

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: scripts/deploy/package-memorypipeline.sh [hunk 1]
```diff
@@ -86,7 +86,16 @@ if [[ ! -d "$PUBLISH_ZIP_DIRECTORY" ]]; then
...
-dotnet publish "$SCRIPT_ROOT/../../memorypipeline/CopilotChatMemoryPipeline.csproj" --configuration $CONFIGURATION --framework $DOTNET --runtime $RUNTIME --self-contained --output "$PUBLISH_OUTPUT_DIRECTORY" /p:AssemblyVersion=$VERSION /p:FileVersion=$VERSION /p:InformationalVersion=$INFO
+dotnet publish "$SCRIPT_ROOT/../../memorypipeline/CopilotChatMemoryPipeline.csproj" \
+    --configuration $CONFIGURATION \
+    --framework $DOTNET \
+    --runtime $RUNTIME \
+    --self-contained \
+    --output "$PUBLISH_OUTPUT_DIRECTORY" \
+    //p:AssemblyVersion=$VERSION \
+    //p:FileVersion=$VERSION \
+    //p:InformationalVersion=$INFO
```
Comment (glahaye) [inline scripts/deploy/package-memorypipeline.sh:97]: This should not be commented out.

If this is what is giving us problems, maybe we just need to surround the arguments with quotes? (Ex: "$VERSION"

Comment (TaoChenOSU) [reply scripts/deploy/package-memorypipeline.sh:97]: These are not comments. Bash scripts use `#` for comments. An additional `/` is needed here to escape the following `/`.

Comment (glahaye) [reply scripts/deploy/package-memorypipeline.sh:97]: Ah!  I see, it's so Bash doesn't interpret what follows the / as a path...

I'm still a bit dazed by this ,but OK! :)

**Review by glahaye — APPROVED** [review]

**Review by dehoward — APPROVED** [review]

**Comment by Rsimk** [comment]
was this actually tested on Linux/WSL?
It still gives the error :(
```
/chat-copilot/scripts/deploy$ ./package-webapi.sh
Building backend executables...
Build configuration: Release
MSBuild version 17.7.3+8ec440e68 for .NET
MSBUILD : error MSB1008: Only one project can be specified.
    Full command line: '/usr/share/dotnet/sdk/7.0.402/MSBuild.dll -maxcpucount -verbosity:m -target:Restore --property:_IsPublishing=true -property:PublishDir=/home/xxxxxxxx/repo/xxxxxxxx/chat-copilot/scripts/deploy/publish -property:_CommandLineDefinedOutputPath=true -property:SelfContained=True -property:_CommandLineDefinedSelfContained=true -property:RuntimeIdentifier=linux-x64 -property:_CommandLineDefinedRuntimeIdentifier=true -property:Configuration=Release /home/xxxxxxxx/repo/xxxxxxxx/chat-copilot/scripts/deploy/../../webapi/CopilotChatWebApi.csproj //p:AssemblyVersion=0.0.0 //p:FileVersion=0.0.0 //p:InformationalVersion= -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,/usr/share/dotnet/sdk/7.0.402/dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,/usr/share/dotnet/sdk/7.0.402/dotnet.dll'
  Switches appended by response files:
Switch: //p:AssemblyVersion=0.0.0

For switch syntax, type "MSBuild -help"
```

**Pull request merged**